flink-end-to-end-tests/flink-end-to-end-tests-common/src/main/java/org/apache/flink/tests/util/FlinkDistribution.java

```diff
@@ -0,0 +1,219 @@
+		opt = flinkDir.resolve("opt");
+		lib = flinkDir.resolve("lib");
+		conf = flinkDir.resolve("conf");
+		log = flinkDir.resolve("log");
+	}
+
+	@Override
+	protected void before() throws IOException {
+		defaultConfig = new UnmodifiableConfiguration(GlobalConfiguration.loadConfiguration(conf.toAbsolutePath().toString()));
+		final Path originalConfig = conf.resolve(FLINK_CONF_YAML);
+		final Path backupConfig = conf.resolve(FLINK_CONF_YAML_BACKUP);
+		Files.copy(originalConfig, backupConfig);
+		filesToDelete.add(new AutoClosablePath(backupConfig));
+	}
+
+	@Override
+	protected void after() {
+		try {
+			stopFlinkCluster();
+		} catch (IOException e) {
+			LOG.error("Failure while shutting down Flink cluster.", e);
+		}
+
+		final Path originalConfig = conf.resolve(FLINK_CONF_YAML);
+		final Path backupConfig = conf.resolve(FLINK_CONF_YAML_BACKUP);
+
+		try {
+			Files.move(backupConfig, originalConfig, StandardCopyOption.REPLACE_EXISTING);
+		} catch (IOException e) {
+			LOG.error("Failed to restore flink-conf.yaml", e);
+		}
+
+		for (AutoCloseable fileToDelete : filesToDelete) {
+			try {
+				fileToDelete.close();
+			} catch (Exception e) {
+				LOG.error("Failure while cleaning up file.", e);
+			}
+		}
+	}
+
+	public void startFlinkCluster() throws IOException {
+		AutoClosableProcess.runBlocking("Start Flink cluster", bin.resolve("start-cluster.sh").toAbsolutePath().toString());
+
+		final OkHttpClient client = new OkHttpClient();
+
+		final Request request = new Request.Builder()
+			.get()
+			.url("http://localhost:8081/taskmanagers")
+			.build();
+
+		Exception reportedException = null;
+		for (int x = 0; x < 30; x++) {
+			try (Response response = client.newCall(request).execute()) {
+				if (response.isSuccessful()) {
+					final String json = response.body().string();
+					final JsonNode taskManagerList = OBJECT_MAPPER.readTree(json)
+						.get("taskmanagers");
+
+					if (taskManagerList != null && taskManagerList.size() > 0) {
+						LOG.info("Dispatcher REST endpoint is up.");
+						return;
+					}
+				}
+			} catch (IOException ioe) {
+				reportedException = ExceptionUtils.firstOrSuppressed(ioe, reportedException);
+			}
+
+			LOG.info("Waiting for dispatcher REST endpoint to come up...");
+			try {
+				Thread.sleep(1000);
+			} catch (InterruptedException e) {
+				Thread.currentThread().interrupt();
+				reportedException = ExceptionUtils.firstOrSuppressed(e, reportedException);
+			}
+		}
+		throw new AssertionError("Dispatcher REST endpoint did not start in time.", reportedException);
+	}
+
+	public void stopFlinkCluster() throws IOException {
+		AutoClosableProcess.runBlocking("Stop Flink Cluster", bin.resolve("stop-cluster.sh").toAbsolutePath().toString());
+	}
+
+	public void copyOptJarsToLib(String jarNamePattern) throws FileNotFoundException, IOException {
+		final Optional<Path> reporterJarOptional = Files.walk(opt)
+			.filter(path -> path.getFileName().toString().startsWith("flink-metrics-prometheus"))
```

Comment:
Here you forgot to replace the `flink-metrics-prometheus` with `jarNamePattern`.

Reply:
yes